src/commands/dev/mod.rs

```diff
@@ -18,23 +18,32 @@ pub fn dev(
     deploy_config: DeployConfig,
     user: Option<GlobalUser>,
     server_config: ServerConfig,
-    http: bool,
+    local_https: bool,
+    upstream_http: bool,
     verbose: bool,
 ) -> Result<(), failure::Error> {
     // before serving requests we must first build the Worker
     build(&target)?;
 
+    if server_config.host.is_https() && upstream_http {
+        failure::bail!("Can't upstream http with https host")
+    } else if local_https && upstream_http {
+        failure::bail!("Can't have local https and upstreamed http")
```

Comment:
```suggestion
        failure::bail("--local-protocol cannot be https if --upstream-protocol is http")
```

maybe suggest using console::style for the arguments here